snapshotrepository_test.go

```diff
@@ -0,0 +1,177 @@
+package eventsourcing_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/hallgren/eventsourcing"
+	"github.com/hallgren/eventsourcing/eventstore/memory"
+	snap "github.com/hallgren/eventsourcing/snapshotstore/memory"
+)
+
+func TestSaveAndGetSnapshot(t *testing.T) {
+	eventrepo := eventsourcing.NewEventRepository(memory.Create())
+	eventrepo.Register(&Person{})
+
+	snapshotrepo := eventsourcing.NewSnapshotRepository(snap.Create(), eventrepo)
+
+	person, err := CreatePerson("kalle")
```

Comment:
For me it is unclear where the setup is, where the code gets tested, and when the result it asserted.

You could perhaps add three sections with comments:
- `// Arrange`: this is where setup is called
- `// Act`: this is where the code is being testing is called
- `// Assert`: this is where we do all the assertions

Reply:
Is it more clear now?